applications/ParticleMechanicsApplication/custom_conditions/particle_based_conditions/mpm_particle_base_condition.h

```diff
@@ -335,7 +335,7 @@ class MPMParticleBaseCondition
      * Calculate Shape Function Values in a given point
      */
 
-    virtual Vector& MPMShapeFunctionPointValues(Vector& rResult, const array_1d<double,3>& rPoint);
+    virtual Vector& MPMShapeFunctionPointValues(Vector& rResult);
```

Comment:
```suggestion
    virtual Vector& MPMShapeFunctionPointValues(Vector& rResult) const;
```

also why pass by reference AND return?

Reply:
I did it like that as e.g. in the penalty condition, because of the small cut error, the value of the shape functions need to be changed. Is there a more elegant way to do this.

Reply:
I would do 
`virtual void MPMShapeFunctionPointValues(Vector& rResult) const;`

Reply:
I changed this accordingly :)